### DISCUSSION TITLE
Can't keep last active cell in state

### DISCUSSION THREAD

**Discussion by Aivanaso** [nick-keller/react-datasheet-grid#194] [Q&A] [answered]
Hi! First of all thank you for this amazing component!

I need to make a function to duplicate the entire selected row, but changing some cells values.

I don't know if I can do this by overriding duplicateRow function, because I need some extra parameters to check whatever cell I should edit, so I'm trying to store last active cell to get the row and adding a new one to the state by getting the current row through **ref.current.activeCell**

```ts
useEffect(() => {
  if (ref.current?.activeCell?.row !== data.activeCell?.row) {
    setData((data) => ({
      ...data,
      activeCell: ref.current?.activeCell ?? null,
    }));
  }
}, [ref.current?.activeCell]);
```

But nothing happens, and I don't know what I'm doing wrong :(

I tried by setting and interval just like this https://github.com/nick-keller/react-datasheet-grid/discussions/139#discussioncomment-2234698 and it's working, but I don't want this way because it's slowing everything.

Can you help me? 

Thanks!!

**Comment by nick-keller** [comment, marked as answer]
Hello @Aivanaso ,
For performance issues, the parent component is NOT re-rendered when the active cell changes, which is why the `useEffect` does not run.

If you want something to happen when the active cell changes, use the [onActiveCellChange](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onactivecellchange)callback prop.

**Reply by Aivanaso** [reply]
Yay! It slows a little bit the change of the active cell in the sheet (maybe because I'm using the dynamic version?) but it's working!

Thanks! :D

**Reply by nick-keller** [reply]
If you are using the dynamic version, make sur to `useMemo` everything. [More info in the doc](https://react-datasheet-grid.netlify.app/docs/performance/static-vs-dynamic#dynamic-version).
